odl/tomo/operators/xray_trafo.py

```diff
@@ -107,35 +112,44 @@ def __init__(self, discr_dom, geometry, backend='astra', **kwargs):
                 raise ValueError('ASTRA backend not available.')
             if not ASTRA_CUDA_AVAILABLE and self.backend == 'astra_cuda':
                 raise ValueError('ASTRA CUDA backend not available.')
-            if discr_dom.dspace.dtype not in (np.float32, np.complex64):
+            if discr_domain.dspace.dtype not in (np.float32, np.complex64):
                 raise ValueError('ASTRA support is limited to `float32` for '
                                  'real and `complex64` for complex data.')
-            if not np.allclose(discr_dom.grid.stride[1:],
-                               discr_dom.grid.stride[:-1]):
+            if not np.allclose(discr_domain.grid.stride[1:],
+                               discr_domain.grid.stride[:-1]):
                 raise ValueError('ASTRA does not support different pixel/voxel'
                                  ' sizes per axis (got {}).'
-                                 ''.format(discr_dom.grid.stride))
+                                 ''.format(discr_domain.grid.stride))
 
         self._geometry = geometry
 
-        # Create discretized space (operator range). Use same data space
+        # Create a discretized space (operator range) with the same data space
         # type as the domain.
         # TODO: maybe use a ProductSpace structure
-        ran_uspace = FunctionSpace(geometry.params)
-        # CHECKME: Is this the right weight?
+        range_uspace = FunctionSpace(geometry.params)
 
         weight = getattr(geometry.grid, 'cell_volume', 1.0)
-
-        ran_dspace = discr_dom.dspace_type(geometry.grid.ntotal,
-                                           weight=weight,
-                                           dtype=discr_dom.dspace.dtype)
-
-        ran_interp = kwargs.pop('range_interpolation', 'nearest')
-        discr_ran = DiscreteLp(ran_uspace, geometry.grid, ran_dspace,
-                               interp=ran_interp, order=geometry.grid.order)
-        super().__init__(discr_dom, discr_ran, linear=True)
-
-        self._adjoint = DiscreteXrayTransformAdjoint(self)
+        if isinstance(geometry, HelicalConeFlatGeometry):
```

Comment:
Another comment. What about the general `DivergentBeam` case, it would give different results from the helical case, even though they should be equivalent. Can we solve that somehow?